### PR TITLE
executor: fix the wrong result when use index_merge on the partition

### LINKED ISSUE #16404: IndexMerge get wrong result
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?
download the file with all DDLs 
[randgen_index_merge.txt](https://github.com/pingcap/tidb/files/4479741/randgen_index_merge.txt)
```
source randgen_index_merge.txt
```    
```
SELECT /*+ use_index_merge(table_20_utf8_4) */ count(*) FROM table_20_utf8_4 where `col_char(20)_key_signed` = "it's" or `col_bigint_key_signed` > -2617317647556123437;
```   
### 2. What did you expect to see?
16
### 3. What did you see instead?
0
### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
Release Version: v4.0.0-rc-33-gd69868961
Git Commit Hash: d69868961361126d4788b68fd6cd597adb3ef66c
Git Branch: release-4.0
UTC Build Time: 2020-04-15 02:59:00
GoVersion: go1.13.8
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

### LINKED ISSUE #16910: unexpected result when use index_merge on partition
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?
``` sql
drop table if exists t1, t2, t3;
create table t1 (a int not null, b tinyint not null, index (a), index (b)) partition by range (a) (
partition p0 values less than (10),
partition p1 values less than (20),
partition p2 values less than (30),
partition p3 values less than (40),
partition p4 values less than MAXVALUE
);
insert into t1 values(0, 0), (1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6), (7, 7), (10, 10), (11, 11), (12, 12), (13, 13), (14, 14), (15, 15), (20, 20), (21, 21), (22, 22), (23, 23), (24, 24), (25, 25), (30, 30), (31, 31), (32, 32), (33, 33), (34, 34), (35, 35), (36, 36), (40, 40), (50, 50), (80, 80), (90, 90), (100, 100);
create table t2 (a int not null, b bigint not null, index (a), index (b)) partition by hash(a) partitions 10;
insert into t2 values (0, 0), (1, 1), (2, 2), (3, 3), (4, 4), (5, 5), (6, 6), (7, 7), (8, 8), (9, 9), (10, 10), (11, 11), (12, 12), (13, 13), (14, 14), (15, 15), (16, 16), (17, 17), (18, 18), (19, 19), (20, 20), (21, 21), (22, 22), (23, 23);

select /*+ USE_INDEX_MERGE(t1, a, b) */ * from t1 partition (p0) join t2 partition (p1) on t1.a = t2.a where t1.a < 40 or t1.b < 30;

```

### 2. What did you expect to see?
``` sql
+---+---+---+---+
| a | b | a | b |
+---+---+---+---+
| 1 | 1 | 1 | 1 |
+---+---+---+---+
```

### 3. What did you see instead?
``` sql
tidb> select /*+ USE_INDEX_MERGE(t1, a, b) */ * from t1 partition (p0) join t2 partition (p1) on t1.a = t2.a where t1.a < 40 or t1.b < 30;
Empty set (0.00 sec)
```

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
``` sql
tidb> select tidb_version();
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                        |
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-beta.2-369-g0e5af57f6
Git Commit Hash: 0e5af57f6088f6de6cb2cedba627922de72aed8f
Git Branch: master
UTC Build Time: 2020-04-28 08:00:59
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve?
Issue Number: close #16910 , close #16404 

Problem Summary:
Forget to set the partition table ID when we use `index_merge_reader`.

### What is changed and how it works?
Set the partition table ID when we use `index_merge_reader` on the partition table.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

### Release note 
- fix wrong result when use index_merge on partition

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -2415,6 +2415,11 @@ func buildTableReq(b *executorBuilder, schemaLen int, plans []plannercore.Physic
 	}
 	ts := plans[0].(*plannercore.PhysicalTableScan)
 	tbl, _ := b.is.TableByID(ts.Table.ID)
+	isPartition, physicalTableID := ts.IsPartition()
```
Comment (lawyerphx) [inline executor/builder.go:2418]: It seems that the same operation will be done in builder:2446. Will it raise some problem?

Comment (Reminiscent) [reply executor/builder.go:2418]: It seems the function `buildNoRangeIndexLookUpReader` will not affect the function `buildTableReq`.

Comment (Reminiscent) [reply executor/builder.go:2418]: Good Catch! I will fix it.

**Pull request merged**